webpack/scenes/Subscriptions/Manifest/ToggleManifestSCA.js

```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Switch } from '@patternfly/react-core';
+
+class SimpleSwitch extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      isChecked: props.simpleContentAccess,
+    };
+    this.handleChange = (isChecked) => {
+      this.setState({ isChecked });
+      this.props.toggleSCA();
+      this.props.hideModal();
+    };
+  }
+
+  render() {
+    const { isChecked } = this.state;
+
+    return (
+      <Switch
+        id="simple-switch"
+        isChecked={isChecked}
+        onChange={this.handleChange}
+        isDisabled={!this.props.canToggleSimpleContentAccess}
+      />
+    );
+  }
```

Comment:
If this is just the Switch, any reason to create a separate component for it?  Seems like this could all be done in the parent component.

If you are concerned about the size of `ManageManifestModal`, I'm all for refactoring to use smaller components or breaking a part out, but I think we should look at larger groups of components to abstract out.

Reply:
My reasoning behind doing it in a separate component is, The responsibility of ManageManifestModal is showing data related to manifest in a modal while doing actions on manifest(e.g. toggling simple content action) should be done in separate component as it has its own state. Any thoughts?

Reply:
I think we could handle the state fine in the parent component, if it gets complex a separate component is fine, I'm just not sure its doing more than shuffling props to Switch, and I don't see it as a reusable component (for example if there was custom CSS to make the switch look different, that could justify breaking it out to be re-used again).

Here is my suggestion for keeping it in `ManageManifesModal`: 
```javascript
// Add to state in constructor
this.state = {
  contentAccessChecked: props.simpleContentAccess
}

// Inside render
<Switch
 id="content-access-toggle-switch"
 isChecked={contentAccessChecked}
 onChange={() => { 
   this.setState({ contentAccessChecked });
   simpleContentAccess ? disableSimpleContentAccess() : enableSimpleContentAccess();
   this.hideModal();
 }}
  // double check my logic here, any of these things would mean the switch is disabled? 
  isDisabled={ disableManifestActions || !manifestExists(organization) || actionInProgress} />
```

If the above achieves the same thing as what you are doing now, I don't see a reason to break this out to another component. I'm open to other opinions from @jturel and @jeremylenz too!

Reply:
Thanks @johnpmitsch for such a detailed explanation. I will do the same as you mentioned.

Reply:
@johnpmitsch what about extracting all the data related to simple content access to newer component(suggested by Jturel? It will refactor the `ManageManifesModal` in a larger group also ?

Reply:
I think that's the way to go: moving the rest of the 'stuff' currently on the ManageManifest modal related to this SCA toggle into the component itself. That way we are much closer to having a truly reusable component that we can drop on any page.

Reply:
+1 to refactoring a larger group of components out of ManageManifestModal, that would be great :+1: